src/types/config.ts

```diff
@@ -83,31 +94,40 @@ export class UpdatableConfig implements Readonly<Config> {
       traceBufferSizeMb: 256,
       maxSearchDepth: 10,
     }
-    this.configureProjectRoot()
+
+    if (isNode()) {
+      this.configureProjectRoot()
+    }
   }
 
   /**
    * Configures the project root by searching for a specific file within a depth limit.
    * This is only supported in a Node environment.
    */
-  private configureProjectRoot() {
+  private async configureProjectRoot() {
+    await loadNodeModules()
+
     // fileURLToPath and dirname is only available in Node, hence the check
+    const { fileURLToPath } = url
+    const { dirname } = path
+
     const _dirname =
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
       // @ts-ignore: Unreachable code error
       // eslint-disable-next-line no-restricted-syntax
-      typeof __dirname !== 'undefined' ? __dirname : fileURLToPath && dirname ? dirname(fileURLToPath(import.meta.url)) : undefined
+      typeof __dirname !== 'undefined' ? __dirname : dirname(fileURLToPath(import.meta.url))
+
     if (!_dirname) {
       return
     }
 
-    let currentPath = resolve(_dirname)
+    let currentPath = path.resolve(_dirname)
     for (let i = 0; i < this.config.maxSearchDepth; i++) {
-      if (existsSync(`${currentPath}/.algokit.toml`)) {
+      if (fs.existsSync(`${currentPath}/.algokit.toml`)) {
```

Comment:
Given this method is now async you could use fs/promises rather than the sync methods if you wanted